airflow/cli/cli_parser.py

```diff
@@ -778,6 +782,12 @@ class GroupCommand(NamedTuple):
         func=lazy_load_command('airflow.cli.commands.dag_command.dag_list_dags'),
         args=(ARG_SUBDIR, ARG_OUTPUT),
     ),
+    ActionCommand(
+        name='generate_kubernetes_pod_yaml',
+        help="Generate YAML files for all tasks in DAG",
+        func=lazy_load_command('airflow.cli.commands.dag_command.generate_pod_yaml'),
+        args=(ARG_SUBDIR, ARG_DAG_ID, ARG_OUTPUT_PATH, ARG_EXEC_DATE),
```

Comment:
```suggestion
        args=(ARG_SUBDIR, ARG_DAG_ID, ARG_OUTPUT_PATH, ARG_EXECUTION_DATE),
```
This will surely make this argument required.

Reply:
@mik-laj  does it need to be required? this is just a diagnostic tool so it shouldn't make a difference if we use a dummy execution_date.

Reply:
This is not necessary, but should we update the other commands as well? Using CLI is very often based on habits, and I think it's worth making sure that all commands have a similar style.
```
airflow dags test example_bash_operator 2019-01-01
airflow dags generate_kubernetes_pod_yaml example_bash_operator --exec-date 2019-01-01
```
I suspect these two commands may be used by the same people and follow two different conventions. This is very confusing to the user. I am in favor of making this argument optional, but it should be done in all commands, or none of them.

Reply:
IMHO execution_date should never be a positional argument. However making this change would be way outside of the scope of this ticket. Can we merge this as is and open a thread on the mailing list for a wider discussion of this?

Reply:
Is there any reason why the command `airflow dags generate_kubernetes_pod_yaml` should have exeuction_date optional, and the command `airflow dags test` has this argument required? In my opinion, these two commands have a similar role. These are commands created for development.